src/axom/quest/detail/MeshTester_detail.hpp

```diff
@@ -0,0 +1,447 @@
+#ifdef AXOM_USE_UMPIRE
+  static constexpr bool ExecOnDevice =
+    axom::execution_space<ExecSpace>::onDevice();
+  static constexpr MemorySpace Space =
+    ExecOnDevice ? axom::MemorySpace::Device : axom::MemorySpace::Host;
+  static constexpr MemorySpace HostSpace = axom::MemorySpace::Host;
+#else
+  static constexpr MemorySpace Space = axom::MemorySpace::Dynamic;
+  static constexpr MemorySpace HostSpace = axom::MemorySpace::Dynamic;
+#endif
+
+  /*!
+   * \brief Initializes the CandidateFinder query object with a surface mesh.
+   *
+   * \param [in] surface_mesh The triangular surface mesh to query.
+   * \param [in] intersectionThreshold The tolerance threshold to use for
+   *  triangle intersection tests.
+   */
+  CandidateFinderBase(mint::UnstructuredMesh<mint::SINGLE_SHAPE>* surface_mesh,
+                      double intersectionThreshold)
+    : m_surfaceMesh(surface_mesh)
+    , m_intersectionThreshold(intersectionThreshold)
+  { }
+
+  void initialize();
+
+  /*!
+   * \brief Runs a query to find pairs of triangle cell indices intersecting
+   *  in the surface mesh, as well as degenerate triangles in the mesh.
+   *
+   * \param [in] firstIndex The first indices of intersecting pairs
+   * \param [in] secondIndex The second indices of intersecting pairs
+   * \param [in] degenerateIndices Indices of degenerate mesh triangles
+   */
+  void findTriMeshIntersections(axom::Array<IndexType>& firstIndex,
+                                axom::Array<IndexType>& secondIndex,
+                                axom::Array<IndexType>& degenerateIndices);
+
+protected:
+  /*!
+   * \brief Returns the candidate intersection pairs by using a spatial query
+   *  data structure.
+   *
+   * \param [out] offsets Offsets into the candidates array for each mesh cell
+   * \param [out] counts  The number of candidates for each mesh cell
+   * \return candidates The flat array of candidate indices
+   *
+   * \note Upon completion, the triangular mesh element at index i has:
+   *  * counts[ i ] candidates
+   *  * Candidate intersections with indices stored in the range:
+   *    [offsets[i], offsets[i] + counts[i])
+   * \note This should be implemented in derived CandidateFinder template
+   *  specializations for each supported acceleration data structure.
+   */
+  virtual axom::ArrayView<IndexType, 1, Space> getCandidates(
+    axom::Array<IndexType, 1, Space>& offsets,
+    axom::Array<IndexType, 1, Space>& counts) = 0;
+
+  mint::UnstructuredMesh<mint::SINGLE_SHAPE>* m_surfaceMesh;
+  double m_intersectionThreshold;
+  int m_ncells;
+  axom::Array<detail::Triangle3, 1, Space> m_tris;
+  axom::Array<BoxType, 1, Space> m_aabbs;
+  axom::Array<int, 1, Space> m_degenerate;
+};
+
+template <typename ExecSpace, typename FloatType>
+void CandidateFinderBase<ExecSpace, FloatType>::initialize()
+{
+  const int ncells = m_surfaceMesh->getNumberOfCells();
+
+  m_tris.resize(ncells);
+  axom::ArrayView<detail::Triangle3, 1, Space> p_tris = m_tris;
```

Comment:
Would this be easier to understand if it were written as the following equivalent:
```suggestion
  auto p_tris = m_tris.view();
```

(I can see both sides of this...)